core/dbt/contracts/connection.py

```diff
@@ -215,6 +215,7 @@ def to_target_dict(self):
 class QueryComment(dbtClassMixin):
     comment: str = DEFAULT_QUERY_COMMENT
     append: bool = False
+    job_label: bool = False
```

Comment:
The config is `query-comment`, rather than `query_comment` ([docs](https://docs.getdbt.com/reference/project-configs/query-comment)). Not exactly sure why we did this, but I guess kebab casing is common in `dbt_project.yml` / project-level configs. I think this should be `job-label` (instead of `job_label`) for consistency

Reply:
OK, I switched the base class to `HyphenatedDbtClassMixin`.

Reply:
On second thought, I may not understand what this base class does, since that change broke tests that I thought were unrelated. Reverted for now, but let me know if you have thoughts about the right way to do this.

Reply:
I took another look at this, and I think I found a subtle bug in the casing logic. Should be fixed now.